guide/creating-your-bot/command-handling.md

```diff
@@ -83,110 +97,258 @@ discord-bot/
 
 	try {
-		await command.execute(interaction);
+		command.execute(interaction);
 	} catch (error) {
 		console.error(error);
 		await interaction.reply({ content: 'There was an error while executing this command!', ephemeral: true });
 	}
 });
+
+client.login(token);
+```
+:::
+::::
+
+First, we create a new property called `commands` for our `client` object, and initialize it as an empty `Collection`. (line 7)
+
+Next, just like in our `deploy-commands.js` file, we will also make use of `readdirSync()` to import all of our command files. However, this time we will call [`Collection.set()`](https://discord.js.org/#/docs/collection/main/class/Collection?scrollTo=set) to populate the `client.commands`, with the command's name as the key, and the command file itself as the value.
+
+Next, we can go to the callback of the `interactionCreate` event, and call `.get()` on the `client.commands` Collection, to retrieve the corresponding command file by its command name. (line 23)
+
+Finally, we call the `.execute()` function of the command file, which will effectively reply to the command.
+
+::: tip
+The <DocsLink section="collection" path="class/Collection" /> from `discord.js` is a class that extends JavaScript's native [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) class, and includes more extensive, useful functionality.
+:::
+
+## Summary
+
+The goal of a *command handler* is to allow each command to be in its own separate file. Not only will this keep the `deploy-commands.js` and `index.js` files cleaner, but it will also allow each command to have its "`data`" (structure) and "`execute`" in the same file; making it easier to develop the `execute` function.
+
+Take the `commands/ping.js` file as example:
+
+:::: code-group
+::: code-group-item commands/ping.js
+```js {4-6,8-10}
+const { SlashCommandBuilder } = require('@discordjs/builders');
+
+module.exports = {
+	data: new SlashCommandBuilder()
+		.setName('ping')
+		.setDescription('Replies with pong!'),
+
+	async execute(interaction) {
+		await interaction.reply('Pong!');
+	}
+};
+```
+:::
+::::
+We have the `data` of our `ping` command, which defines the "structure" of the command. It defines how the command will look when a user tries to use the command in Discord.
+
+Below it, we have the `execute` of our `ping` command, which defines how we want our bot to reply to the `ping` command.
+
+The `deploy-commands.js` file will import the `data` (structure) of the command for registration:
+
+:::: code-group
+::: code-group-item deploy-commands.js
+```js :no-line-numbers {2}
+const command = require(`./commands/${file}`);
+commands.push(command.data.toJSON());
+```
+:::
+::::
+
+while the `index.js` file will import the file and then call the `execute()` of the command:
+
+:::: code-group
+::: code-group-item index.js
+```js :no-line-numbers {2,4,6}
```

Comment:
i can see your point and i thought the same as well. but i ended up including `no-line-numbers` because the line number will not match up with the actual file, and therefore might create confusion for the reader.